gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/ProductDependency.java

```diff
@@ -45,22 +45,36 @@
     @JsonProperty("maximum-version")
     private String maximumVersion;
 
+    @JsonProperty("optional")
+    private boolean optional = false;
```

Comment:
I think we need a test that you can't set an `optional` pdep on a jar using the `recommended-product-dependencies` plugin (as we don't want optional pdeps on jars, at least not yet). I think we can add a check [after this line](https://github.com/palantir/sls-packaging/blob/138001775a7f3a16a539e4d8050efb32f88c46b9/gradle-recommended-product-dependencies/src/main/java/com/palantir/gradle/dist/RecommendedProductDependenciesExtension.java#L43) that throws if the `optional` field is set.

Reply:
Done